### PR TITLE
Query : PeerList

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Adds support for query `PeerList`

## Changes
Adds support for query `PeerList`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.concordium.sdk.responses.peerList;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java:1]:
```suggestion
package com.concordium.sdk.responses.peerlist;
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import inet.ipaddr.IPAddressString;
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * Represents the parsed form of {@link ConcordiumP2PRpc.PeerElement}
+ */
+@Builder
+@Getter
+public class Peer {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java:15]: Please add `toString` for convenience.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    private final String nodeId;
+    private final UInt16 port;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java:18]: Please add `toString` on the `Uint16.java`.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/PeerCatchupStatus.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * Node is downloading new updates.
+     */
+    CATCHINGUP(2),
+
+    /**
+     * Unknown status.
+     */
+    UNRECOGNIZED(-1);
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/PeerCatchupStatus.java:28]: Please remove this and throw an `IllegalArgumentException` with a sufficient error message in `parse` below.

File: concordium-sdk/pom.xml [hunk 5]
```diff
@@ -71,6 +71,11 @@
             <artifactId>jackson-datatype-jsr310</artifactId>
             <version>2.13.2</version>
         </dependency>
+        <dependency>
```
Comment (EmilLa1) [inline concordium-sdk/pom.xml:74]: What is the reason for not just using `InetAddress`?

If no particular reason exists, then I would prefer if we just use `InetAddress` and avoid having a dependency for this logic.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good 👍 
Only a few minor comments.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java (hunk 1 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java (hunk 2 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/Peer.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/PeerCatchupStatus.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerList/PeerCatchupStatus.java]: done

File: concordium-sdk/pom.xml (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/pom.xml]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerlist/PeerCatchupStatus.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+     * @return Parsed {@link PeerCatchupStatus}.
+     */
+    public static PeerCatchupStatus parse(ConcordiumP2PRpc.PeerElement.CatchupStatus catchupStatus) {
+        switch (catchupStatus) {
+            case UPTODATE: return PeerCatchupStatus.UPTODATE;
+            case PENDING: return PeerCatchupStatus.PENDING;
+            case CATCHINGUP: return PeerCatchupStatus.CATCHINGUP;
+            default:
+            throw new IllegalArgumentException(String.format(
+                    "Invalid catchup status received : %d",
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerlist/PeerCatchupStatus.java:43]:
```suggestion
                    "Invalid catchup status received: %d",
```

File: README.md [hunk 7]
```diff
@@ -195,6 +195,11 @@ long getTotalSent()
 ```
 Retrives total number of packets sent from the node.
 
+- `getPeerList`
+```java
+ImmutableList<Peer> getPeerList(boolean includeBootstrappers) throws UnknownHostException
+```
+
```
Comment (EmilLa1) [inline README.md:202]:
```suggestion
Retrieves the peers that the node is connected to.
The boolean flag `includeBootstrappers` indicates whether bootstrapper nodes 
should be included in the response.
```

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good! 👍 

Just two very minor comments regarding documentation.

**Review by EmilLa1 — APPROVED** [review]
Looks good to go.

**Pull request merged**